src/iot/commands/createPolicy.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs-extra'
+import { getLogger } from '../../shared/logger'
+import { Commands } from '../../shared/vscode/commands'
+import { Window } from '../../shared/vscode/window'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { showViewLogsMessage } from '../../shared/utilities/messages'
+import { IotPolicyFolderNode } from '../explorer/iotPolicyFolderNode'
+
+/**
+ * Creates a policy from a policy document.
+ */
+export async function createPolicyCommand(
+    node: IotPolicyFolderNode,
+    window = Window.vscode(),
+    commands = Commands.vscode()
+): Promise<void> {
+    getLogger().debug('CreatePolicy called for %O', node)
+
+    const policyName = await window.showInputBox({
+        prompt: localize('AWS.iot.createPolicy.prompt', 'Enter a new policy name'),
+        placeHolder: localize('AWS.iot.createPolicy.placeHolder', 'Policy Name'),
+    })
+
+    if (!policyName) {
+        getLogger().info('CreatePolicy canceled')
+        return
+    }
+
+    const fileLocation = await promptForPolicyLocation(window)
+    if (!fileLocation) {
+        getLogger().info('CreatePolicy canceled: No document selected')
+        return
+    }
+
+    try {
+        const data = await fs.readFile(fileLocation.fsPath)
```

Comment:
Similar situation here, except for reading a file. Is there a readFile mock or a temporary test file that can be used?

Reply:
>  a temporary test file that can be used?

You would need to make your own fixture potentially, but it seems like the logic here is indifferent of the contents.

> I was thinking I could factor out this dependency into a separate function, and pass that function is in a parameter to allow it to be overridden in the unit test. 

Seems reasonable to me. You could combine the prompt with reading the file.